barretenberg/cpp/src/barretenberg/plonk_honk_shared/types/aggregation_object_type.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+
+namespace bb {
+constexpr uint32_t AGGREGATION_OBJECT_SIZE = 16;
+using AggregationObjectIndices = std::array<uint32_t, AGGREGATION_OBJECT_SIZE>;
+using AggregationObjectPubInputIndices = std::array<uint32_t, AGGREGATION_OBJECT_SIZE>;
```

Comment:
Previously, this was a vector but it should be an array of fixed size

Reply:
the next step would be to turn each one into a class and have a flag that says whether it is set or not